ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx

```diff
@@ -74,7 +74,7 @@ export function NotificationsSettingsAllowNotifications({
   }, [isMetamaskNotificationsEnabled]);
 
   useEffect(() => {
-    if (!error) {
+    if (!error && isMetamaskNotificationsEnabled) {
```

Comment:
Add additional check to only run content if enabled.

Reply:
I mentioned in the description that I dislike the multiple ~useFootgun~ useEffect's spread around in a bunch of components. I'll do a larger refactor and cleanup.